### PR TITLE
feat(benchmarking): tune benchmarking for non-return syscalls

### PR DESCRIPTION
**Pull request by tltsutltsu**
Resolves: #2938

There are a couple of benchmarks preceded with this comment:
```
// We cannot call `gr_wait` multiple times. Therefore our weight determination is not 
// as precise as with other APIs. 
```
 
Lets call them "one-time benchmarks" 
 
This PR resolves this problem by splitting the benchmarking process of `pallet_gear` into two runs. The first run is made with default parameters (`--steps=20`, `--repeat=50`) for all extrinsics, excluding one-time ones. The second run is made only for one-time extrinsics, with `--steps=2`, `--repeat=1000` params. This allows to evaluate one-time benchmarks more thoroughly.

Here's how it's done, technically:
- `run_all_benchmarks.sh` obtains a list of all extrinsics.
- `run_all_benchmarks.sh` runs all benchmarks, excluding one-time ones. The output is written into the `pallet_gear.rs` file.
- `run_all_benchmarks.sh` runs only one-time benchmarks. The output is written into the `pallet_gear_onetime.rs` file.
- `merge-outputs.sh` moves `WeightInfo` trait definitions and weights from the `pallet_gear_onetime.rs` file to main `pallet_gear.rs` file.

No changes needed in CI, and everything, including weights unpacking script, is are already working fine.
[Successful benchmark action run](https://github.com/gear-tech/gear/actions/runs/6234875598/job/16922930562) / [Weights diff table](https://github.com/gear-tech/gear/pull/3319#issuecomment-1731378626)

The set of these one-time benchmarks can be changed anytime without need of modifying the logic in the script. Additionally , `merge-outputs.sh` also allows to merge more than one file by specifying multiple files.

@gear-tech/dev

### CODE REVIEW

**Comment by ukint-vs** [comment]
Could you post a table here with the weight changes?

File: scripts/benchmarking/run_all_benchmarks.sh [hunk 1]
```diff
@@ -10,6 +10,33 @@
...
+# List of one-time extrinsics to benchmark. This is only used for pallet_gear.
+ONE_TIME_EXTRINSICS=(
+  "gr_reply"
+  "gr_reply_wgas"
+  "gr_reply_commit"
+  "gr_reply_commit_wgas"
+  "gr_reply_input"
+  "gr_reply_input_wgas"
+  "gr_reservation_reply"
+  "gr_reservation_reply_commit"
```
Comment (StackOverflowExcept1on) [inline scripts/benchmarking/run_all_benchmarks.sh:31]: `gr_reservation_reply_commit` is duplicated here

maybe we can get `ONE_TIME_EXTRINSICS` using the `grep`?
```
cat pallets/gear/src/benchmarking/mod.rs | grep "0 .. 1;" -B 1 | grep -E "{$" | awk '{print $1}'
```

**Comment by tltsutltsu** [comment]
@ukint-vs 
> Could you post a table here with the weight changes?

Sure!

| name                                    | master   | tuned | diff    |
|-----------------------------------------|----------|------------------------------------------|---------|
| gr_reply_commit                         | 20.8 µs  | 16.7 µs                                                      | +24.61% |
| gr_reply_commit_wgas                    | 16.0 µs  | 23.4 µs                                                      | -31.69% |
| gr_reservation_reply                    | 11.1 µs  | 10.2 µs                                                      | +8.71%  |
| gr_reservation_reply_commit             | 5.3 µs   | 10.3 µs                                                      | -48.34% |
| gr_reply                                | 16.8 µs  | 24.5 µs                                                      | -31.20% |
| gr_reply_wgas                           | 20.5 µs  | 21.7 µs                                                      | -5.37%  |
| gr_reply_input                          | 21.9 µs  | 20.2 µs                                                      | +8.30%  |
| gr_reply_input_wgas                     | 0 ps     | 30.9 µs                                                      | N/A     |
| gr_exit                                 | 23.7 µs  | 181.1 µs                                                     | -86.92% |
| gr_leave                                | 12.2 µs  | 172.9 µs                                                     | -92.95% |
| gr_wait                                 | 13.2 µs  | 149.0 µs                                                     | -91.17% |
| gr_wait_for                             | 14.0 µs  | 185.5 µs                                                     | -92.46% |
| gr_wait_up_to                           | 12.2 µs  | 167.1 µs                                                     | -92.68% |

These weights were measured with 1000 repeats.

[Here is a full table with other weights](https://gist.github.com/tltsutltsu/6c6c6403cd454338fc52d4859e712f39)

File: scripts/benchmarking/run_all_benchmarks.sh (hunk 1 above)
Comment (tltsutltsu) [reply scripts/benchmarking/run_all_benchmarks.sh]: Super cool, thx!

**Review by ukint-vs — APPROVED** [review]

**Review by breathx — APPROVED** [review]

**Pull request merged**